### PR TITLE
Govsi 519 interruption screen for terms and conds update

### PR DESCRIPTION
**Pull request by cj1993**
## What?

Front end changes to support updating the terms and conditions from the interrupt screen pop up

### CODE REVIEW

File: src/components/updated-terms-conds/updated-terms-conds-controller.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+): ExpressRouteFunc {
+  return async function (req: Request, res: Response) {
+    const updatedTermsAndCondsValue = true;
+    const {email} = req.session.user;
+    const sessionId = res.locals.sessionId;
+    const redirectUri = req.session.redirectUri;
+    const acceptOrReject = req.body.acceptOrReject;
+
+    if (acceptOrReject === "reject") {
+      res.redirect(redirectUri);
```
Comment (sanisc) [inline src/components/updated-terms-conds/updated-terms-conds-controller.ts:31]: i think we want to redirect to the service with an error message so would be something like:
`res.redirect(redirectUri+ "?error_code=rejectedTermsAndConditions");`

File: src/components/updated-terms-conds/updated-terms-conds-routes.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { PATH_NAMES } from "../../app.constants";
+
+import * as express from "express";
+import {updatedTermsCondsGet, updatedTermsCondsPost} from "./updated-terms-conds-controller";
+import {validateSessionMiddleware} from "../../middleware/session-middleware";
+import {asyncHandler} from "../../utils/async";
+
+const router = express.Router();
+
+router.get(PATH_NAMES.UPDATED_TERMS_AND_CONDITIONS, asyncHandler(updatedTermsCondsGet()));
```
Comment (sanisc) [inline src/components/updated-terms-conds/updated-terms-conds-routes.ts:10]: the validatesessionmiddleware should be used on the get

File: src/components/updated-terms-conds/updated-terms-conds-controller.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Request, Response } from "express";
```
Comment (sanisc) [inline src/components/updated-terms-conds/updated-terms-conds-controller.ts:1]: there is no unit tests or integration tests for this - can we get them added

File: src/components/updated-terms-conds/index.njk [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+          tag: {
+            text: 'general.phaseBanner.tag' | translate
+          },
+          html: 'This is a new service – your <a class="govuk-link" href="#">feedback</a> will help us to improve it.'
+        }) }}
+	      {% block beforeContent %}{% endblock %}
+	      <main class="govuk-main-wrapper {{ mainClasses }}" id="main-content" role="main"{% if mainLang %} lang="{{ mainLang }}"{% endif %}>
+          <div class="govuk-grid-row interrupt-screen">
+            <div class="govuk-grid-column-two-thirds">
+              {% extends "common/layout/base.njk" %}
```
Comment (sanisc) [inline src/components/updated-terms-conds/index.njk:60]: how come this page is being imported with the base layout inside the main content block?

**Review by sanisc — CHANGES_REQUESTED** [review]

File: src/components/updated-terms-conds/updated-terms-conds-controller.ts (hunk 3 above)
Comment (cj1993) [reply src/components/updated-terms-conds/updated-terms-conds-controller.ts:1]: Threw unit tests in, the integration tests have been a bit of a pain. May have to revisit them with another dev if we can get this PR sorted.

**Review by sanisc — APPROVED** [review]

**Pull request merged**